torchci/pages/tests/search.tsx

```diff
@@ -0,0 +1,106 @@
+import { Pagination, Stack, Tooltip } from "@mui/material";
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import LoadingPage from "components/LoadingPage";
+import TestSearchForm from "components/tests/TestSearchForm";
+import { useRouter } from "next/router";
+import { ListTestInfoAPIResponse } from "pages/api/flaky-tests/search";
+import { useEffect, useState } from "react";
+import useSWR from "swr";
+
+const fetcher = (url: string) => fetch(url).then((res) => res.json());
+
+export function encodeParams(params: { [key: string]: string }) {
+  return Object.keys(params)
+    .map((key) => `${key}=${encodeURIComponent(params[key])}`)
+    .join("&");
+}
+
+export default function Page() {
+  const router = useRouter();
+  const name = (router.query.name || "") as string;
+  const suite = (router.query.suite || "") as string;
+  const file = (router.query.file || "") as string;
+  const [page, setPage] = useState(1);
+  const [count, setCount] = useState(1);
+  const { data, isLoading } = useSWR<ListTestInfoAPIResponse>(
+    `/api/flaky-tests/search?${encodeParams({
+      name,
+      suite,
+      file,
+      per_page: "100",
+      page: page.toString(),
+    })}`,
+    fetcher
+  );
+
+  const columns: GridColDef[] = [
+    {
+      field: "name",
+      headerName: "Name",
+      flex: 2,
+      renderCell: (params) => {
+        return (
+          <Tooltip title={params.value}>
+            <span>{params.value}</span>
+          </Tooltip>
+        );
+      },
+    },
+    { field: "classname", headerName: "Classname", flex: 1 },
+    { field: "file", headerName: "File", flex: 1 },
+    { field: "invoking_file", headerName: "Invoking File", flex: 1 },
+    { field: "last_run", headerName: "Last Run", flex: 1 },
+  ];
+
+  useEffect(() => {
+    if (data) {
+      setCount(Math.ceil(data.count / 100));
+    }
+  }, [data]);
+
+  if (!router.isReady) {
+    // router.query is not set on the initial render, so I return this in order
+    // to force the entire component to re-render once the query is set.  This
+    // gets around the TextField defaultValue getting set to "" and then not
+    // changing when the router changes.  Another option could be setting state
+    // but that is a lot of variables to set
+    return <LoadingPage />;
+  }
+
+  return (
+    <Stack spacing={{ xs: 1 }}>
+      <h1>Test Search</h1>
+      <TestSearchForm name={name} suite={suite} file={file} />
+      <div>Double click a row to see individual stats</div>
```

Comment:
IMO, if you could turn the test name into a link, it will be pretty intuitive that people can click there to view the details and there is no need for this line.

An example is https://hud.pytorch.org/benchmark/torchbench/inductor_aot_inductor?dashboard=torchinductor&startTime=Mon%2C%2020%20Jan%202025%2022%3A06%3A06%20GMT&stopTime=Mon%2C%2027%20Jan%202025%2022%3A06%3A06%20GMT&granularity=hour&mode=inference&dtype=bfloat16&deviceName=cuda%20(a100)&lBranch=main&lCommit=e90cf4abcff190bb97cd365eeee7196c697b749f&rBranch=main&rCommit=b5655d9821b7214af200d0b8796a10ad34b85229 where the model name is a link pointing to the model details